### PR TITLE
Create a template to look like github's repo preview

### PR DESCRIPTION
**Pull request by jckoester**
I've created a first template for github repositories to look quite a lot like github's own  repo preview.
If you like the idea, I could also create templates for the remaining embeds.

I could also imagine to add an option to select bewteen a more classic view (using the github octocat icon) and a more github like.

### CODE REVIEW

File: css/github-embed.css [hunk 1]
```diff
@@ -22,4 +22,82 @@ li.github-repo-contributor {
...
 	margin-right: 1em;
 	margin-bottom: 0.5em;
+}
+
+.github-embed-repository{
+    padding-top: 16px!important;
+    padding-bottom: 16px!important;
+    border-bottom: 1px #e1e4e8 solid !important
+}
+.d-flex{
```
Comment (leewillis77) [inline css/github-embed.css]: I'm not keen on redeclaring styles with names that obviously conflict with well-known frameworks such as bootstrap-4. Ideally I'd like to see these either be prefixed so they don't conflict, or (my personal preference) use real semantic classes for things and define the styles we want against those, e.g.

File: github-embed.php [hunk 2]
```diff
@@ -273,33 +273,23 @@ private function oembed_github_repo ( $owner, $repository ) {
...
-				$cnt++;
-			}
-			$response->html .= '</ul>';
-		}
-		$response->html .= '</p>';
-		$response->html .= '</div>';
-		header( 'Content-Type: application/json' );
+                // Include template file
+                // @TODO: there should be an option to select different templates!
+                $includepath = plugin_dir_path( __FILE__ ) . 'templates/repo_default.php';
```
Comment (leewillis77) [inline github-embed.php]: If we're going to start using template files (we totally should!) then it should be done properly. This should pick up the template from the theme directories (using locate_template() or similar) if it exists before falling back to a template shipped with the plugin. That gives people the flexibility they'd expect and makes it easy to customise if they want to.

File: github-embed.php [hunk 3]
```diff
@@ -273,33 +273,23 @@ private function oembed_github_repo ( $owner, $repository ) {
...
+                
+                // Start rendering
+		ob_start();
```
Comment (leewillis77) [inline github-embed.php]: What's the purpose of the output buffering here? I can't see that it's actually achieving anything?

File: github-embed.php [hunk 4]
```diff
@@ -273,33 +273,23 @@ private function oembed_github_repo ( $owner, $repository ) {
...
-				$response->html .= esc_html( $commit->commit->committer->name );
-				$response->html .= '</li>';
-				$cnt++;
-			}
-			$response->html .= '</ul>';
-		}
-		$response->html .= '</p>';
-		$response->html .= '</div>';
-		header( 'Content-Type: application/json' );
+                // Include template file
```
Comment (leewillis77) [inline github-embed.php]: The indentation is off in this whole block.

File: templates/repo_default.php [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+             </a>
+            <?php } ?>
+
+            Updated on <?php echo date( "j M Y", strtotime( esc_attr( $repo->updated_at ) ) ); ?>
+        </div>
+      </div>
+    </div>
+    
+
+<?php //echo "<pre>"; ?>
```
Comment (leewillis77) [inline templates/repo_default.php]: Please can you remove any debug before resubmitting (even commented out debug).

Thanks

**Comment by dasmaeh** [comment]
Don't worry about your response time, mine is worse.

Thanks for your comments, I'll do the changes requested. However it will take some time.

File: github-embed.php (hunk 3 above)
Comment (dasmaeh) [reply github-embed.php]: Output buffering allows the contents of a 'normal' template file to be rendered into a variable. If you have a better solution to this, please let me know.

Comment (leewillis77) [reply github-embed.php]: You're right. Ignore me :)

**Comment by leewillis77** [comment]
Closed since the plugin now supports templating.

**Pull request closed without merging**